### PR TITLE
Met2model LPJ-GUESS

### PR DESCRIPTION
**Pull request by istfer**
met2model function for LPJ-GUESS #753 

some notes:
(i) LPJ-GUESS reads CF-standard climate data, only looks for each climate variable in a separate input file. So this function reads PEcAn-CF files which have all the different climate variables in one file for each year, and re-writes separate netcdf files for each variable that have all the climate data for all years: 
reads CF2000,CF2001....,CF2009,CF2010 ---> writesTMP2000-2010, PRE2000-2010, CLD2000-2010

(ii) LPJ-GUESS can also take wetdays, min/max-temp, but min requirement is temperature, precipitation and cloudiness. So this function only handles those. CF-units are OK.

(iii) LPJ-GUESS requires at least 30 years of data to build spinup and I had everything set up for 1901-2006 period before. So I tested the function with CRUNCEP data covering that period to see whether LPJ-GUESS would actually work with the function outputs. But 3 things came up:
- [the function to load CRUNCEP data](https://github.com/PecanProject/pecan/blob/master/modules/data.atmosphere/R/download.CRUNCEP_Global.R) cuts one data point (I end up with time dimension of 1459 for a normal year whereas it should be 1460 (=365x4)). @mdietze should there be an issue for this?
- I think that function downloads the files even if they're there, some if-skip block would be good as in [download.NARR](https://github.com/PecanProject/pecan/blob/master/modules/data.atmosphere/R/download.NARR.R#L47)

```
      if (file.exists(new.file) && !overwrite) {
        logger.debug("File '", new.file, "' already exists, skipping to next file.")
        next
      }
```
(iv) The third thing: The main problem I had was making LPJ-GUESS read the climate data for 1 grid (actually even for 2 and 3 grids) because once it read the data back from the netcdf file it loses dimensions (i.e. instead of having a tmp[lat,lon,time] 3D-array it ends up with a tmp[time] vector. Right now I'm writing the same climate data, for same grid 4-times to work aroud it. Is there a way to solve it at the netcdf-writing step (force it to preserve dimensions or something like that?)

(v) For now met2model passes through the workflow (special thanks to @tonygardella for helping me to solve NARR/BETY puzzle), but LPJ-GUESS still runs its static version. In other words no matter what you define in the pecan.xml (except a different insfile path) it will run the 1901-2006 period. My next step is to modify it through the workflow.

looking forward to @crollinson's feedback!

### CODE REVIEW

File: models/lpjguess/R/met2model.LPJGUESS.R [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    ## open netcdf files
+    ncin <- lapply(file.path(in.path,paste(in.prefix,year,"nc",sep=".")),nc_open)
+    
+    ## retrieve lat/lon 
+    lon=ncvar_get(ncin[[1]],"longitude")
+    lat=ncvar_get(ncin[[1]],"latitude")
+    
+    ## at least 2 lat-lon required for LPJ-GUESS to load the data
+    lon=c(lon,lon)
```
Comment (istfer) [inline models/lpjguess/R/met2model.LPJGUESS.R]: this and writing the same climate data four times for all (same) four grids below are the hacks that I could have come up with to make it work for LPJ-GUESS

File: models/lpjguess/R/met2model.LPJGUESS.R [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  for(i in 1:n.var) out.file[[i]] <-paste(in.prefix, start_year, end_year, var.names[[i]], "nc", sep=".")
+  for(i in 1:n.var) out.files.full[[i]] <-file.path(outfolder,out.file[[i]])
+  
+  results <- data.frame(file = unlist(out.files.full),
+                        host = fqdn(),
+                        mimetype ='application/x-netcdf',
+                        formatname = 'lpj-guess.metfile' ,
+                        startdate = start_date ,
+                        enddate = end_date,
+                        dbfile.name = unlist(out.file),
```
Comment (istfer) [inline models/lpjguess/R/met2model.LPJGUESS.R]: I haven't seen an example in the other met2model functions where we write 3 files: how is this going to be used later? is passing three file names OK here?

Comment (dlebauer) [inline models/lpjguess/R/met2model.LPJGUESS.R]: In principle it should be fine. eg you could pass a directory name where the three files are. Or in BioCro names of met and soil input files are given separately.

**Comment by mdietze** [comment]
👍 from me

File: models/lpjguess/R/met2model.LPJGUESS.R [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    ## read climate data
+    nc.tmp <- lapply(ncin, ncvar_get, long.names[1])
+    nc.pre <- lapply(ncin, ncvar_get, long.names[2])
+    nc.cld <- lapply(ncin, ncvar_get, long.names[3])
+    
+    ## aggregate to daily time steps, LPJ-GUESS reads daily climate data
+    tmp.list <- pre.list <- cld.list <- list()
+    for(y in 1:nyear){
+      if(leap_year(as.numeric(year[y]))) ind.vec=rep(1:366,each=tstep) else ind.vec=rep(1:365,each=tstep)
+      tmp.list[[y]] <- tapply(nc.tmp[[y]],ind.vec,mean)
```
Comment (crollinson) [inline models/lpjguess/R/met2model.LPJGUESS.R]: I think here is where you might be losing the 3 dimensions LPJ-GUESS needs to run.  I think you could either set up each level of tmp.list (etc) as an array with dims c(1, 1, time) or do in in your ncvar_put statement (I'll comment there too)

**Comment by crollinson** [comment]
after chatting with @istfer about my suggestions to get the dims working right, we couldn't figure it out, so this with the hack looks good to me.

**Pull request merged**